src/socket/tcp.rs

```diff
@@ -311,6 +314,30 @@ impl<'a> TcpSocket<'a> {
         }
     }
 
+    /// Return the time-to-live (IPv4) or hop limit (IPv6) value used in outgoing packets.
+    ///
+    /// See also the [set_ttl](#method.set_ttl) method
+    pub fn ttl(&self) -> Option<u8> {
+        self.ttl
+    }
+
+    /// Set the time-to-live (IPv4) or hop limit (IPv6) value used in outgoing packets.
+    ///
+    ///  * A socket without an explicitly set TTL value uses the default [INNA recommended]
+    ///    value (`64`).
+    ///  * A TTL value of 0 is invalid and will be ignored. See [RFC 1122 § 3.2.1.7].
+    ///
+    /// [INNA recommended]: https://www.iana.org/assignments/ip-parameters/ip-parameters.xhtml
+    /// [RFC 1122 § 3.2.1.7]: https://tools.ietf.org/html/rfc1122#section-3.2.1.7
+    pub fn set_ttl(&mut self, ttl: Option<u8>) {
+        // A host MUST NOT send a datagram with a Time-to-Live (TTL)
+        // value of 0
+        match ttl {
+            Some(0)  => {},
+            catchall => self.ttl = catchall,
+        }
```

Comment:
Realized it was possible to set the TTL value for sent packets to the invalid value of 0 with `set_ttl`. Added logic to ensure that the sockets configured TTL value is only changed when the given argument is `None` or `Some(x)` where `x > 0`